src/main/java/uk/nhs/prm/e2etests/queue/SimpleAmqpQueue.java

```diff
@@ -95,7 +95,7 @@ private Producer createProducer() {
             }
         }
 
-        throw new GenericException(this.getClass().getName(), "Failed to connect to any endpoint");
+        throw new GenericException(this.getClass().getName(), "Failed to connect to any endpoint, this can occur when you are not connected to the VPN. If you are connected to the VPN, try restart the AMQP broker.");
```

Comment:
Possibly scope creep, though if we can move away from `GenericException` to something more explicit - such as `AmqpException`. The reflection seems overkill too.

Reply:
I completely agree with your points. It's definitely scope creep in this context, there's a tech debt item to address all uses of 'GenericException'.